### PR TITLE
DOC: remove experimental tag from CL

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Constrained layout is not really experimental any more, and we probably should not break it at this point....

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by jklymak** [comment]
Note I also adjusted the `Figure` layout docstring.  I think both layout managers suffer performance constraints, just the worse ones for constrained_layout you simply wouldn't even expect anything sensible from tight_layout.

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -2212,22 +2212,23 @@ def __init__(self,
...
                 ``layout='constrained'`` instead.
 
-        layout : {'constrained', 'tight'}, optional
-            The layout mechanism for positioning of plot elements.
-            Supported values:
+        layout : {'constrained', 'tight'}, optional, default: None
+            The layout mechanism for positioning of plot elements to avoid
+            overlapping Axes decorations (labels, ticks, etc). Note that
+            layout managers can have significant performance penalties as
+            they require an extra draw and solve an inverse problem.
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply lib/matplotlib/figure.py:2219]: I'm weakly in favor of brevity here as well.

**Review by tacaswell — APPROVED** [review]
Same as @dstansby 

@jklymak can self-merge with or without the wording change.

File: lib/matplotlib/figure.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jklymak) [reply lib/matplotlib/figure.py:2219]: Done...

**Pull request merged**